CMakeLists.txt

```diff
@@ -0,0 +1,713 @@
+# accuracy, completeness,  or usefulness of  any information, apparatus,
+# product, or  process disclosed, or  represents that its use  would not
+# infringe privately-owned rights.
+# 
+# Any reference herein to  any specific commercial products, process, or
+# services by trade name,  trademark, manufacturer or otherwise does not
+# necessarily  constitute or imply  its endorsement,  recommendation, or
+# favoring  by  the  United  States  Government  or  Lawrence  Livermore
+# National Security,  LLC. The views  and opinions of  authors expressed
+# herein do not necessarily state  or reflect those of the United States
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+###
+# grab the version string
+###
+file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/VERSION SILO_VERSION)
+
+###-----------------------------------------------------------------------------
+# project command will automatically create cmake vars for major, minor,
+# patch and tweak from passed VERSION 
+###-----------------------------------------------------------------------------
+project(Silo VERSION ${SILO_VERSION} LANGUAGES CXX C)
+
+###-----------------------------------------------------------------------------
+# location for Silo CMake includes 
+###-----------------------------------------------------------------------------
+set(CMAKE_MODULE_PATH ${Silo_SOURCE_DIR}/CMake)
+
+###-----------------------------------------------------------------------------
+# If not already set, use a default build type of Release
+###-----------------------------------------------------------------------------
+
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build,
+options are: Debug Release RelWithDebInfo MinSizeRel.")
+endif()
+
+###-----------------------------------------------------------------------------
+# Output directories.
+###-----------------------------------------------------------------------------
+
+if(NOT DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY lib CACHE INTERNAL "Output directory for libraries.")
+endif()
+
+if(NOT DEFINED CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY lib CACHE INTERNAL "Output directory for archives.")
+endif()
+
+if(NOT DEFINED CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY bin CACHE INTERNAL "Output directory for executables.")
+endif()
+
+# turn on folders (VS IDE, ignored otherwise)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+#
+#  NOTE Currently doing everything from this one CMakeLists.txt
+#  So everything is relative to as ${Silo_SOURCE_DIR} which will be the same
+#  as ${CMAKE_CURRENT_SOURCE_DIR} unless some things are moved into subdirs.
```

Comment:
Not sure I understand the part about `...unless some things are moved into subdirs` here.